dev/doc/xml-protocol.md

```diff
@@ -151,18 +151,19 @@ state that should become the next tip.
 * Failure:
   - Syntax error. Error offsets are byte offsets (not character offsets) with respect to the start of the sentence, starting at 0.
   ```html
-  <value val="fail"
-      loc_s="${startOffsetOfError}"
-      loc_e="${endOffsetOfError}">
+  <value val="fail">
     <state_id val="${stateId}"/>
+    <option val="none|some">${loc}</option>
     <richpp>${errorMessage}</richpp>
   </value>
   ```
   - Another kind of error, for example, Qed with a pending goal.	
   ```html
-  <value val="fail"><state_id val="${stateId}"/><richpp>${errorMessage}</richpp></value>
+  <value val="fail"><state_id val="${stateId}"/><option val="some">${loc}</option><richpp>${errorMessage}</richpp></value>
   ```
 
+  Note that locations may need to be adjusted by the IDE to character offsets in the presence of multi-byte characters. Also, due to asynchronous evaluation, a location may have to adjusted if the sentence has moved since it was sent to Coqtop.
```

Comment:
```suggestion
  Note that IDEs may need to convert byte offsets passed in the four position fields of the location
  to character offsets to correctly handle multi-byte characters. Also, due to asynchronous
  evaluation, line number fields of locations may need to be adjusted if the sentence has moved
  since it was sent to Coqtop.
```

more detail, slight wording improvement and split into multiple lines